lib/storage.js

```diff
@@ -209,6 +209,60 @@ exports.set = function(key, json, callback) {
 };
 
 /**
+ * @summary Write multiple sets of user data
+ * @function
+ * @public
+ *
+ * @param {Object} keys
+ * @param {Function} callback - callback (error, done)
+ *
+ * @example
+ * const storage = require('electron-json-storage');
+ *
+ * storage.setMany({ foo: 'bar', bar: 'foo' }, function(error, done) {
+ *  if (error) {
+ *    console.log('Error saving ' + error.key);
+ *    throw error.error;
+ *  }
+ *  if (finished) console.log('finished');
+ * });
+ */
+exports.setMany = function(keys, callback) {
+  if (!keys || keys.constructor !== Object) {
+    callback(new Error('Invalid keys'), true);
+    return;
+  }
+
+  const entries = _.map(keys, function(value, key) {
+    return { key: key, value: value };
+  });
+
+  function process() {
+    const entry = entries.shift();
+    const key = entry.key;
+    const value = entry.value;
+
+    exports.set(key, value, function(error) {
+      if (error) {
+        callback({ error: error, key: key }, entries.length === 0);
```

Comment:
That was originally what I had planned. With the way I ended up doing it, the user will still know which ones to retry (it will set the keys one by one, and every time there is an error it will invoke the callback which is why the callback takes a second parameter `finished` to let the user know when all the keys have been set).

It's up to you. If we did it your way, would we simply return a list of failed keys (`['foo', 'bar']`) or the errors that they threw too (`{ foo: error1, bar: error2 }`)?